hooks/src/index.js

```diff
@@ -239,8 +244,10 @@ function safeRaf(callback) {
 if (typeof window !== 'undefined') {
 	let prevRaf = options.requestAnimationFrame;
 	afterPaint = (component) => {
-		if ((!component._afterPaintQueued && (component._afterPaintQueued = true) && afterPaintEffects.push(component) === 1) ||
-		    prevRaf !== options.requestAnimationFrame) {
+		if (
+			(!component._afterPaintQueued && (component._afterPaintQueued = afterPaintEffects.push(component)) === 1)
```

Comment:
Question: In core's state queue, we always push onto the state array and just dedupe the resulting array when flushing (invoking the callback sets a boolean that makes subsequent invocations inert). I wonder if that'd be cheaper here?

Reply:
Oh interesting... Yea that probs is.

Though in the next PR I plan on opening I rewrote this mechanism to no longer need the `_afterPaintQueued` at all. [Relevant commit is here](https://github.com/preactjs/preact/commit/650e93f5436d75c17d3ab7f13a7c7ecc107956d6) if you are curious. So I'll leave this for now and we can perhaps discuss again in the next PR. Thoughts?

Reply:
Gonna leave it for now. PR #2012 has the change where we no longer need to track if the effect has been queued.